signature-aggregator/aggregator/aggregator.go

```diff
@@ -0,0 +1,507 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package aggregator
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"math/big"
+	"math/rand"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/message"
+	"github.com/ava-labs/avalanchego/proto/pb/p2p"
+	"github.com/ava-labs/avalanchego/subnets"
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/utils/set"
+	avalancheWarp "github.com/ava-labs/avalanchego/vms/platformvm/warp"
+	"github.com/ava-labs/awm-relayer/peers"
+	"github.com/ava-labs/awm-relayer/utils"
+	coreEthMsg "github.com/ava-labs/coreth/plugin/evm/message"
+	msg "github.com/ava-labs/subnet-evm/plugin/evm/message"
+	"go.uber.org/zap"
+)
+
+type blsSignatureBuf [bls.SignatureLen]byte
+
+const (
+	// Number of retries to collect signatures from validators
+	maxRelayerQueryAttempts = 5
+	// Maximum amount of time to spend waiting (in addition to network round trip time per attempt)
+	// during relayer signature query routine
+	signatureRequestRetryWaitPeriodMs = 10_000
+)
+
+var (
+	codec        = msg.Codec
+	coreEthCodec = coreEthMsg.Codec
+
+	// Errors
+	errNotEnoughSignatures     = errors.New("failed to collect a threshold of signatures")
+	errNotEnoughConnectedStake = errors.New("failed to connect to a threshold of stake")
+)
+
+type SignatureAggregator struct {
+	network                 *peers.AppRequestNetwork
+	subnetIDsByBlockchainID map[ids.ID]ids.ID
+	logger                  logging.Logger
+	messageCreator          message.Creator
+	currentRequestID        atomic.Uint32
+	mu                      sync.RWMutex
```

Comment:
Major nit: could we name this `requestIDLock`? I believe elsewhere in our codebases we always name `sync.Mutex`s something ending in "lock"

Reply:
Actually I removed the requestIDLock since it's just uint32 and made it atomic instead. The lock above protects `subnetsByBlockchainID` but the point stands